### PR TITLE
Set up default product license for community engines

### PR DESCRIPTION
**Pull request by dhiltgen**
This will report a fixed string for community engines

Related to https://github.com/moby/moby/pull/37612

### CODE REVIEW

File: deb/Makefile [hunk 1]
```diff
@@ -16,13 +16,15 @@ COMMON_FILES=common
 BUILD?=docker build \
 	--build-arg GO_IMAGE=$(GO_IMAGE) \
 	--build-arg COMMON_FILES=$(COMMON_FILES) \
+	--build-arg DEFAULT_PRODUCT_LICENSE="$(DEFAULT_PRODUCT_LICENSE)" \
```
Comment (seemethere) [inline deb/Makefile]: Let's just make a `PRODUCT_LICENSE` variable that we can override and have a default value set in all of the Makefiles.

Comment (dhiltgen) [reply deb/Makefile]: So you'd like a:
```make
DEFAULT_PRODUCT_LICENSE?="Community Engine"
```
in every makefile instead of just the top-level makefile?

Comment (dhiltgen) [reply deb/Makefile]: I've made the suggested change.

**Comment by dhiltgen** [comment]
@seemethere I've sorted out the build glitch related to quotes, so I think this should be ready for another look and merge if you're happy with it.

**Comment by dhiltgen** [comment]
I realized it was ~pointless to pass these into the package builds now since we're just using the image based build as the payload.  While I was cleaning that up and testing it I realized other parameters weren't being passed in properly either, so I fixed those too.

@seemethere ptal.

File: image/Makefile [hunk 2]
```diff
@@ -27,8 +28,12 @@ $(ENGINE_DIR)/Dockerfile.engine:
 # utilize manifests
 image-linux: $(ENGINE_DIR)/Dockerfile.engine
 	docker build -t $(DOCKER_HUB_ORG)/$(ENGINE_IMAGE):$(STATIC_VERSION).$(ARCH) \
-	    --build-arg VERSION=$(STATIC_VERSION) \
-	    --build-arg DOCKER_GITCOMMIT=$$(cd $(ENGINE_DIR) && git rev-parse --short=7 HEAD) \
+	    --build-arg VERSION="$(STATIC_VERSION)" \
+	    --build-arg GITCOMMIT="$$(cd $(ENGINE_DIR) && git rev-parse --short=7 HEAD)" \
+	    --build-arg BUILDTIME="$(BUILDTIME)" \
```
Comment (seemethere) [inline image/Makefile]: Where is builtime coming from in this case?

Comment (dhiltgen) [reply image/Makefile]: Good question - I was trying to wire this up using the same variables from the prior build rigging, but apparently that's buried somewhere in "hacks" I guess.

+1 on a follow up PR to mop up any additional straggler.

**Pull request merged**